src/components/orders/StatusLabel/index.tsx

```diff
@@ -110,19 +177,35 @@ function StatusIcon({ status }: DisplayProps): JSX.Element {
   return <StyledFAIcon icon={icon} spin={isOpen} />
 }
 
-export type Props = DisplayProps & { partiallyFilled: boolean }
+export type Props = {
+  status: OrderStatus
+  partiallyFilled: boolean
+  filledPercentage?: BigNumber
+  partialTagPosition?: PartiallyTagPosition
+}
 
-export function StatusLabel(props: Props): JSX.Element {
-  const { status, partiallyFilled } = props
+export function StatusLabel({
+  status,
+  partiallyFilled,
+  filledPercentage,
+  partialTagPosition = 'bottom',
+}: Props): JSX.Element {
   const shimming = status === 'signing' || status === 'cancelling'
+  const customizeStatus = status === 'expired' || status === 'cancelled'
+  const tagPartiallyFilled = partiallyFilled && canBePartiallyFilled(status)
+  const formattedPercentage = filledPercentage !== undefined && formatPercentage(filledPercentage)
+  const _status: CustomOrderStatus = customizeStatus && partiallyFilled ? 'partially filled' : status
 
   return (
-    <Wrapper>
-      <Label status={status} shimming={shimming}>
-        <StatusIcon status={status} />
-        {status}
+    <Wrapper
+      partialFill={tagPartiallyFilled}
+      tagPosition={partialTagPosition}
+      filledPercentage={formattedPercentage || 'Partially filled'}
+    >
+      <Label status={_status} shimming={shimming} partialFill={tagPartiallyFilled} tagPosition={partialTagPosition}>
+        <StatusIcon status={_status} />
+        {_status.charAt(0).toUpperCase() + _status.slice(1)}
```

Comment:
Nitpick: Any reason not to use `_status[0]` instead?

Also, this doesn't feel right. Could you create an util function for capitalizing the string?
I'm sure we created them before, just not sure in which repo it was.

Reply:
I agree, I reused `capitalize` function.